pkl-commons-cli/src/main/kotlin/org/pkl/commons/cli/commands/BaseOptions.kt

```diff
@@ -136,6 +136,24 @@ class BaseOptions : OptionGroup() {
       .path()
       .multiple()
 
+  val proxy: URI? by
+    option(
+        names = arrayOf("--proxy"),
+        metavar = "<address>",
+        help = "Proxy to use for HTTP(s) connctions."
+      )
+      .single()
+      .convert { URI(it) }
+
+  val noProxy: List<String>? by
+    option(
+        names = arrayOf("--no-proxy"),
+        metavar = "<pattern1,pattern2>",
+        help = "Hostnames that should not be connected to via a proxy."
+      )
+      .single()
+      .split(",")
```

Comment:
I think it'd be good to only allow this flag if `--proxy` is also set. Otherwise, it can be confusing why `pkl eval --no-proxy "*"` isn't doing anything.

```suggestion
      .split(",")
      .validate { 
        require(proxy != null) {
          "--no-proxy can only be specified alongside a --proxy"
        }
      }
```

Reply:
Rolling this back after offline conversation; proxy config can come from other sources than CLI